Mesh.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+ @file      Mesh.cpp
+ @brief     contains functions for the Mesh class
+ @author    Luke Eure
+ @date      January 9 2016
+*/
+
+#include "Mesh.h"
+
+/*
+ @brief constructor for Mesh class
+*/
+Mesh::Mesh(Boundaries bounds, double delta_x, double delta_y, double delta_z,
+        Material default_material, int num_groups) {
+    
+    // save deltas 
+    _delta_axes.push_back(delta_x);
+    _delta_axes.push_back(delta_y);
+    _delta_axes.push_back(delta_z);
+
+    // save number of groups
+    _num_groups = num_groups;
+
+    // save boundary mins
+    for (int axis=0; axis<3; ++axis) {
+        _boundary_mins.push_back(bounds.getSurfaceCoord(axis, 0));
+    }
+
+    // save axis sizes
+    for (int axis=0; axis<3; ++axis) {
+        int size = (bounds.getSurfaceCoord(axis, 1) - _boundary_mins[axis])
+            / _delta_axes[axis];
+        _axis_sizes.push_back(size);
+    }
+    
+    // resize _flux and set all its elements = 0
+    _flux.resize(_num_groups);
+    for (int i=0; i<_num_groups; ++i) {
+        _flux[i].resize(_axis_sizes[0]);
+        for (int j=0; j<_axis_sizes[0]; ++j) {
+            _flux[i][j].resize(_axis_sizes[1]);
+            for (int k=0; k<_axis_sizes[1]; ++k) {
+                for (int l=0; l<_axis_sizes[2]; ++l) {
+                    _flux[i][j][k].push_back(0.0);
+                }
+            }
+        }
+    }
+
+    // create materials array
+    _cell_materials.resize(_axis_sizes[0]);
+    for (int i=0; i<_axis_sizes[0]; ++i) {
+        _cell_materials[i].resize(_axis_sizes[1]);
+        for (int j=0; j<_axis_sizes[1]; ++j) {
+            for (int k=0; k<_axis_sizes[2]; ++k) {
+                _cell_materials[i][j].push_back(default_material);
```

Comment:
I'm getting an error when I try to resize _cell_materials[i][j]:

Mesh.cpp: In member function ‘void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type, std::vector<_Tp, _Alloc>::value_type) [with _Tp = Material; _Alloc = std::allocator<Material>; std::vector<_Tp, _Alloc>::size_type = long unsigned int; std::vector<_Tp, _Alloc>::value_type = Material]’:
Mesh.cpp:56:56: error: no matching function for call to ‘Material::Material()’
             _cell_materials[i][j].resize(_axis_sizes[2]);